lua/orgmode/org/mappings.lua

```diff
@@ -1022,7 +1022,20 @@ function OrgMappings:_change_todo_state(direction, use_fast_access)
     elseif direction == 'prev' then
       next_state = todo_state:get_prev()
     elseif direction == 'reset' then
-      next_state = todo_state:get_todo()
+      local repeat_to_state = headline:get_property('REPEAT_TO_STATE')
```

Comment:
Let's create a method on `TodoState` called `get_reset_todo`, and pass the headline inside.
Then we can figure out all the logic in there for this:

```lua
next_state = todo_state:get_reset_todo(headline)
```


Inside, we should do two things:
1. Check if headline has property as you do here, if yes, goto step 4
2. if no property on the headline, check if config has `org_todo_repeat_to_state` ( add this with default `nil`, also a documentation update would be nice)
3. If no config option, use the one that we use now
4. Check if value is valid by looking into `todo_keywords.keys[todo_value]`

Reply:
It should be ok now, let me know if there's something more I can do!